superbit_lensing/shear_profiles/annular_jmac.py

```diff
@@ -42,40 +71,38 @@ def __init__(self,cat_info = None, annular_info = None):
         self.r = None
         self.mu = None
 
-        """
-        self.xc = annular_info['nfw_center'][0]
-        self.yc = annular_info['nfw_center'][1]
-        self.startrad = annular_info['shear_args'][0]
-        self.endrad = annular_info['shear_args'][1]
-        self.nbins = annular_info['nbins']
-        """
+        return
 
+    def open_table(self, cat_info):
 
-    def open_table(self,cat_info):
+        annular_file = cat_info['infile']
+        print(f'loading annular file {annular_file}')
 
         try:
-            tab = Table.read(cat_info['incat'],format='ascii')
+            tab = Table.read(annular_file, format='fits')
         except:
-            tab = Table.read(cat_info['incat'],format='fits')
+            tab = Table.read(annular_file, format='ascii')
 
         try:
             self.x = tab[cat_info['xy_args'][0]]
             self.y = tab[cat_info['xy_args'][1]]
             self.g1 = tab[cat_info['shear_args'][0]]
             self.g2 = tab[cat_info['shear_args'][1]]
             #self.mu = tab['nfw_mu']
-        except:
-            print("Could not load xy/g1g2 columns; check supplied column names?")
-            pdb.set_trace()
+        except Exception as e:
+            print('Could not load xy/g1g2 columns; check supplied column names?')
+            raise e
 
         return
 
-    def get_r_gtan(self,write=False):
+    def transform_shears(self, outdir, overwrite=False):
         """
         Populates self.r with radial distance of galaxies from (xc,yc)
         and self.gtan/gcross with tangential and cross ellipticities
 
         Failed shape measurements with g1/g2 = -999999 get filtered out
+
+        confusingly, outfile 
```

Comment:
Sincerely, 

@sweverett

Reply:
It looks like there is an output being saved now. Is there something else missing?

Reply:
I have no idea what that line was supposed to say 😆 

This part got confusing as there were lots of output tables getting written at various stages, and we're now calling it all by 1 script call. I decided that files that get piped to future modules can be defined by an `outfield`, but it was ok if intermediate files had some default name like this one

Reply:
That makes sense.